### PR TITLE
feat!: Use did-resolver v3

### PR DESCRIPTION
**Pull request by oed**
This is a breaking change because the api of the `did-resolver` package has changed. Namely what the format of the result object looks like and the types used for the resolver registry.

Also generated the docs.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -217,18 +217,17 @@ export class DID {
     if (typeof jws !== 'string') jws = fromDagJWS(jws)
     const kid = base64urlToJSON(jws.split('.')[0]).kid as string
     if (!kid) throw new Error('No "kid" found in jws')
-    const { publicKey } = await this.resolve(kid)
+    const didResolutionResult = await this.resolve(kid)
+    const publicKeys = didResolutionResult.didDocument!.verificationMethod || []
```
Comment (PaulLeCam) [inline src/index.ts:221]: As noted by the typecheck warning below, it seems unsafe as `didDocument` could be `null` according to this type: https://github.com/decentralized-identity/did-resolver/blob/master/src/resolver.ts#L17

**Review by PaulLeCam — COMMENTED** [review]
LGTM apart from the typecheck warning raised, any reason it is ignored?

File: src/index.ts (hunk 1 above)
Comment (oed) [reply src/index.ts:221]: It will really only be `null` if there is an error, which is caught in the `this.resolve` method. Maybe I can improve it there?

Comment (oed) [reply src/index.ts]: Updated to use `?` instead.

**Review by PaulLeCam — APPROVED** [review]

**Pull request merged**